saleor/checkout/migrations/0055_create_checkout_metadata_model_and_move_checkout_metadata.py

```diff
@@ -0,0 +1,134 @@
+    checkouts_with_meta = (
+        Checkout.objects.filter(
+            Q(metadata__isnull=False) | Q(private_metadata__isnull=False)
+        )
+        .order_by("pk")
+        .distinct("token")
+    )
+    for batch_pks in queryset_in_batches(checkouts_with_meta):
+        checkouts = Checkout.objects.filter(pk__in=batch_pks)
+        CheckoutMetadata.objects.bulk_create(
+            CheckoutMetadata(
+                checkout=checkout,
+                metadata=checkout.metadata if hasattr(checkout, "metadata") else {},
+                private_metadata=checkout.private_metadata
+                if hasattr(checkout, "private_metadata")
+                else {},
+            )
+            for checkout in checkouts
+        )
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("checkout", "0054_alter_checkout_options"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="CheckoutMetadata",
+            fields=[
+                (
+                    "id",
+                    models.AutoField(
+                        auto_created=True,
+                        primary_key=True,
+                        serialize=False,
+                        verbose_name="ID",
+                    ),
+                ),
+                (
+                    "private_metadata",
+                    models.JSONField(
+                        blank=True,
+                        default=dict,
+                        encoder=saleor.core.utils.json_serializer.CustomJsonEncoder,
+                        null=True,
+                    ),
+                ),
+                (
+                    "metadata",
+                    models.JSONField(
+                        blank=True,
+                        default=dict,
+                        encoder=saleor.core.utils.json_serializer.CustomJsonEncoder,
+                        null=True,
+                    ),
+                ),
+            ],
+            options={
+                "abstract": False,
+            },
+        ),
+        migrations.AddField(
+            model_name="checkoutmetadata",
+            name="checkout",
+            field=models.OneToOneField(
+                on_delete=django.db.models.deletion.CASCADE,
+                related_name="metadata_storage",
+                to="checkout.checkout",
+            ),
+        ),
+        migrations.RunPython(move_all_checkout_metadata, migrations.RunPython.noop),
+        migrations.RemoveIndex(
+            model_name="checkout",
+            name="checkout_p_meta_idx",
+        ),
+        migrations.RemoveIndex(
+            model_name="checkout",
+            name="checkout_meta_idx",
+        ),
+        migrations.RemoveField(
+            model_name="checkout",
+            name="metadata",
+        ),
+        migrations.RemoveField(
+            model_name="checkout",
+            name="private_metadata",
+        ),
```

Comment:
Maybe I'm too precautious about this, but what do you think about not removing these fields yet, but instead treating them just as deprecated, and removing them after we release and everyone migrates to the new solution? I'm just thinking about having a backup option in case anything went wrong.

Reply:
@fowczarek What do you think?

Reply:
Migrations will be changed to use `SeparateDatabaseAndState`. So the fields will be removed from state only and within the next release after this one - removed from DB as well in another migration. Is it acceptable?